node-notes-cli/app.js

```diff
@@ -0,0 +1,57 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+let data = await readFile('data.json', 'utf8');
+data = JSON.parse(data);
+const notesObj = data.notes;
+
+async function rewrite() {
+  await writeFile('data.json', JSON.stringify(data, null, 2));
+}
+
+async function read() {
+  for (const [key, value] of Object.entries(notesObj)) {
+    console.log(`${key}: ${value}`);
+  }
+}
+
+async function add() {
+  const id = data.nextId;
+  data.nextId++;
+  notesObj[id] = process.argv[3];
```

Comment:
You reuse `process.argv[3]` a lot. I understand it's a challenge to extract it into a variable since its usage depends on which command you are executing. Maybe if you put it into a local variable at the top of each function???